File: examples/governance_static_go/domain/updated_review_date.go
package domain

import "time"

type UpdatedReviewDate struct {
	AggregateID string    `json:"aggregate_id"`
	AggregateId string `json:"aggregate_id"`
	PolicyId string `json:"policy_id"`
	ReviewDate time.Time `json:"review_date"`
	Name string `json:"name"`
	Description string `json:"description"`
	Category string `json:"category"`
	FrameworkId string `json:"framework_id"`
	EffectiveDate time.Time `json:"effective_date"`
	Requirements []Requirement `json:"requirements"`
	Status string `json:"status"`
	OccurredAt time.Time `json:"occurred_at"`
}

func (e UpdatedReviewDate) EventName() string { return "UpdatedReviewDate" }

func (e UpdatedReviewDate) GetOccurredAt() time.Time { return e.OccurredAt }

File: examples/governance/compliance_domain/lib/compliance_domain/governance_policy/events/updated_review_date.rb
module ComplianceDomain
  class GovernancePolicy
    module Events
      class UpdatedReviewDate
        attr_reader :aggregate_id, :policy_id, :review_date, :name, :description, :category, :framework_id, :effective_date, :requirements, :status, :occurred_at

        def initialize(aggregate_id: nil, policy_id: nil, review_date: nil, name: nil, description: nil, category: nil, framework_id: nil, effective_date: nil, requirements: nil, status: nil)
          @aggregate_id = aggregate_id
          @policy_id = policy_id
          @review_date = review_date
          @name = name
          @description = description
          @category = category
          @framework_id = framework_id
          @effective_date = effective_date
          @requirements = requirements
          @status = status
          @occurred_at = Time.now
          freeze
        end
      end
    end
  end
end
